### PR TITLE
DEV-27677 I can delete users in bulk

### PR DESCRIPTION
**Pull request by mrnyax**
- Add API to delete users in bulk

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -3761,6 +3761,199 @@ noted in the Create User API section.
...
+                "type": "validation",
+                "title": "Invalid request attributes",
+                "validationDetails": [
+                {
+                    "title": "Validation error",
+                    "constraint": "NotNull",
+                    "detail": "must not be null",
+                    "invalidValue": null
+                }
+                ],
```
Comment (george-haddad) [inline apiary.apib:3925]: Check the indentation here. It looks like the stuff inside the `{  }` should be indented 1 level in.

Comment (mrnyax) [reply apiary.apib]: Fixed it.

File: apiary.apib [hunk 2]
```diff
@@ -3761,6 +3761,199 @@ noted in the Create User API section.
...
+
+    + Attributes (BulkResultResponse)
+
+    + Body
+
+            // When all users were successfully delete
+            {
+                "results": [
+                    {
+                    "status": 204,
```
Comment (tkrisztian95) [inline apiary.apib:3797]: I think these result fields between the `{  ...  }` should be tabulated a bit more right.

File: apiary.apib [hunk 3]
```diff
@@ -3761,6 +3761,199 @@ noted in the Create User API section.
...
+                    "id": "b08309c7-87ee-4270-8d94-de5a916d24be",
+                    "location": null
+                    }
+                ],
+                "errors": []
+                }
+
+
+            // when 4 users were deleted successfully but 1 wasn’t
```
Comment (tkrisztian95) [inline apiary.apib:3826]: I think this comment is not in sync with the provided example response. I would expect `4` items in the `results` array and only `1` in the `errors`

File: apiary.apib [hunk 4]
```diff
@@ -3761,6 +3761,199 @@ noted in the Create User API section.
...
+            "error": {
+                "reference": "a2616dcb-1009-47c6-a6cd-34fe71776dd5",
+                "detail": "Check the request for invalid values or missing parameters.",
+                "type": "validation",
+                "title": "Invalid request attributes",
+                "validationDetails": [
+                {
+                    "title": "Validation error",
+                    "constraint": "Size",
+                    "detail": "size must be between 1 and 2",
```
Comment (tkrisztian95) [inline apiary.apib:3902]: I think, you should use here the original `size must be between 1 and 100` text and simply change the value of `"invalidValue"` to something like 
` "invalidValue": "[d6601a5b-c74e-4753-8d7a-4a7d15e71149, d7b6d898-79b2-4a8b-9afd-760fdda9c921, ...",`. I mean like using the `...` dots to indicate there are more items expected there.

File: apiary.apib [hunk 5]
```diff
@@ -3761,6 +3761,199 @@ noted in the Create User API section.
...
+
+    + Body
+
+            // When all users were successfully delete
+            {
+                "results": [
+                    {
+                    "status": 204,
+                    "id": "d6601a5b-c74e-4753-8d7a-4a7d15e71149",
+                    "location": null
```
Comment (tkrisztian95) [inline apiary.apib:3799]: I think this field should be omitted from the serialization in the case of containing `null` value. 
I remember I've done it in my MR recently. 

Could you please check if the API is really returning the `location` filed with the `null` value?

File: apiary.apib [hunk 6]
```diff
@@ -3761,6 +3761,199 @@ noted in the Create User API section.
...
+### Delete Users in Bulk [DELETE /api/v1/user/bulk]
+
+This API lets you delete a list of users. All user ids follow the requirements and constraints
+noted in the Delete User API section.
+
+**Note:** You can delete up to 100 users per bulk request, the minimum is 1.
+
++ Request Delete Users in Bulk (application/json)
+
+    This request lets you delete 5 users by specifying user uuids in the request object.
```
Comment (tkrisztian95) [inline apiary.apib:3773]: I think here we should also use uppercase `UUIDs` instead.

File: apiary.apib (hunk 2 above)
Comment (mrnyax) [reply apiary.apib:3797]: That's right. Thanks for spotting that

Comment (mrnyax) [reply apiary.apib:3797]: That's right. Thanks for spotting that

File: apiary.apib (hunk 3 above)
Comment (mrnyax) [reply apiary.apib]: Updated the description to match response

**Review by george-haddad — COMMENTED** [review]
Some comments

File: apiary.apib (hunk 5 above)
Comment (mrnyax) [reply apiary.apib]: Indeed they are not on the response.

File: apiary.apib [hunk 7]
```diff
@@ -3761,6 +3761,243 @@ noted in the Create User API section.
             }
         }
 
+### Delete Users in Bulk [DELETE /api/v1/user/bulk]
+
+This API lets you delete a list of users. All user ids follow the requirements and constraints
+noted in the Delete User API section.
```
Comment (madeline-brennan) [inline apiary.apib:3767]:
```suggestion
noted in the "Delete User API" section.
```

File: apiary.apib [hunk 8]
```diff
@@ -3761,6 +3761,243 @@ noted in the Create User API section.
...
+
+**Note:** You can delete up to 100 users per bulk request, the minimum is 1.
+
++ Request Delete Users in Bulk (application/json)
```
Comment (madeline-brennan) [inline apiary.apib:3771]:
```suggestion
+ Request Delete users in bulk (application/json)
```

**Review by madeline-brennan — CHANGES_REQUESTED** [review]
Just a couple of small tweaks

File: apiary.apib [hunk 9]
```diff
@@ -3761,6 +3761,243 @@ noted in the Create User API section.
...
+                    {
+                        "status": 204,
+                        "id": "b08309c7-87ee-4270-8d94-de5a916d24be"
+                    }
+                ],
+                "errors": []
+                }
+
+
+            // when 2 users were deleted successfully but 3 weren't
```
Comment (tkrisztian95) [inline apiary.apib:3823]: Could you please move this response to a separate error example request?

**Review by madeline-brennan — APPROVED** [review]

File: apiary.apib [hunk 10]
```diff
@@ -3761,6 +3761,266 @@ noted in the Create User API section.
...
+                    },
+                    {
+                        "status": 204,
+                        "id": "b08309c7-87ee-4270-8d94-de5a916d24be"
+                    }
+                ],
+                "errors": []
+                }
+
++ Request when 2 users were deleted successfully but 3 weren't (application/json)
```
Comment (tkrisztian95) [inline apiary.apib:3822]: Naming recommendation: 
```suggestion
+ Request Example with users cannot delete (application/json)
```

File: apiary.apib [hunk 11]
```diff
@@ -3761,6 +3761,266 @@ noted in the Create User API section.
...
+                "b08309c7-87ee-4270-8d94-de5a916d24be"
+            ]
+
++ Response 207 (application/json)
+
+    + Attributes (BulkResultResponse)
+
+    + Body
+
+            // when all users were successfully delete
```
Comment (tkrisztian95) [inline apiary.apib:3795]: I think this comment can be also removed from here.
```suggestion

```

**Review by george-haddad — APPROVED** [review]

File: apiary.apib [hunk 12]
```diff
@@ -3973,6 +3973,266 @@ It lets you ask multiple users to update their user information.
...
+                    },
+                    {
+                        "status": 204,
+                        "id": "b08309c7-87ee-4270-8d94-de5a916d24be"
+                    }
+                ],
+                "errors": []
+                }
+
++ Request Example with users cannot delete (application/json)
```
Comment (george-haddad) [inline apiary.apib:4035]: It should be something like 
```suggestion
+ Request Example with users that cannot be deleted (application/json)
```

**Review by george-haddad — COMMENTED** [review]
Found this one change

File: apiary.apib (hunk 12 above)
Comment (madeline-brennan) [reply apiary.apib:4035]: Would use "can't" instead of cannot

Comment (george-haddad) [reply apiary.apib:4035]: I updated the suggestion change, thanks.

**Pull request merged**